pmd-core/src/main/java/net/sourceforge/pmd/lang/document/FileCollector.java

```diff
@@ -310,21 +368,50 @@ public boolean addFileOrDirectory(Path file) throws IOException {
      * {@link #addFile(Path)} and such. The zip file is registered as
      * a resource to close at the end of analysis.
      */
-    public FileSystem addZipFile(Path zipFile) {
+    @Experimental
+    public void addZipFile(Path zipFile) throws IOException {
```

Comment:
This looks like an incompatible change: return type is removed now and a new checked exception is introduced.

Maybe we should keep the old method and deprecate it and create a new method...

Reply:
I've added back the old method as deprecated and renamed the new one to `addZipFileWithContent`.

Reply:
This new method returns a boolean, to behave like the other "add*" methods, which all return a boolean.